### ISSUE TITLE
Enabling OIDC causes all robot accounts to generate error: failed to verify secret, username: robot$gitlab, error: failed to get oidc user info, error: <QuerySeter> no row

### ISSUE THREAD

**Issue by sharkymcdongles** [goharbor/harbor#15253]
It appears when OIDC is enabled the auth goes through OIDC even if the account is a robot account with no OIDC. This causes the logs to be inundated with this error message:

`2021-07-02T14:55:11Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="a2af154dd57eca5e5a05a8ac4012a311"]: failed to verify secret, username: robot$gitlab, error: failed to get oidc user info, error: <QuerySeter> no row `

Since robot accounts cannot even be linked to OIDC accounts is there some sort of way to stop these log messages and errors?

I am using v2.3.0. OIDC settings are:

```
OIDC Provider Name : azuread
OIDC Endpoint: https://login.microsoftonline.com/CENSORED/v2.0
OIDC Client ID: CENSORED
OIDC Client Secret: CENSORED
Group Claim Name: groups
OIDC Admin Group: CENSORED
OIDC Scope: openid,email,profile,offline_access
Verify Certificate: On
Automatic Onboarding: On
Username Claim: email
```

**Comment by DennisGlindhart** [comment]
I'm experiencing the same on a cleanly installed 2.3. Pretty much same settings as OP.
(+ Also CLI for secret/login from docker doesn't seem to work either - Don't know if related, but could be something general in "auth-selection" logic when OIDC is enabled. )

**Comment by sharkymcdongles** [comment]
I also get random failures where the authorisation header are empty on
docker auth to registry that seem like they might be related and sound
similar to what Dennis wrote. Perhaps they are one and the same.

**Comment by DennisGlindhart** [comment]
Can reproduce on clean version 2.2.2 also, so nothing new for 2.3

**Comment by sharkymcdongles** [comment]
Some more info from debug logs:

```
2021-07-03T11:37:41Z [DEBUG] [/pkg/oidc/secret.go:75]: Verifying the secret for user: robot$gitlab
2021-07-03T11:37:41Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="1135d5b3c15180380de126e5d2617cc9"]: failed to verify secret, username: robot$gitlab, error: failed to get oidc user info, error: <QuerySeter> no row found
2021-07-03T11:37:41Z [ERROR] [/pkg/token/token.go:66]: parse token error, token contains an invalid number of segments
2021-07-03T11:37:41Z [DEBUG] [/server/middleware/security/robot.go:49][requestID="1135d5b3c15180380de126e5d2617cc9"]: failed to decrypt robot token of v1 robot: robot$gitlab, as: token contains an invalid number of segments
```


Seems it tries to verify the robot token via oidc despite it not being oidc and then  it fails to do so and then the token it generates is invalid.

**Comment by Morriz** [comment]
we have the same issue with the harbor v2.3.0-047b122c ;(

**Comment by Morriz** [comment]
I tested on 2.1.5, 2.2.2 and 2.3.0...all suffer this

How are we supposed to pull images again? Isn't this a core part of Harbor?

**Comment by Morriz** [comment]
I tested with an old setup of ours and this issue is not seen, so excuse me for not trying more exhaustively before wining...for now all we can see is the fact that we upgraded istio to 1.10. Will investigate further...

**Comment by sharkymcdongles** [comment]
I wonder if maybe there is some issue with the assigning the auth header
when a reverse proxy is involved like the nginx ingress.

**Comment by Morriz** [comment]
tnx for helping out...I was wondering the same thing. The auth header is set by keycloak and all other internal services that act as client to it have no issues. But it must be related to the header somehow

**Comment by Morriz** [comment]
I now see one cluster that has no issues with exactly the same setup as on a duplicate cluster I created. One is allowing the login now fine, the other isn't. No pods failing, can't pin down why this intermittent issue exists.

**Comment by Morriz** [comment]
so at first I thought the logs were relevant, but the working setup logs exactly the same lines, even though auth works:
```
core 2021-07-05T20:15:40Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="de6b09e5-ce00-4369-a4d6-4a08054a72ce"]: failed to verify secret, username: robot$xxxxxx, error: failed to get oidc user info, error: <QuerySeter> no row found                                                                                            │
core 2021-07-05T20:15:40Z [ERROR] [/pkg/token/token.go:66]: parse token error, token contains an invalid number of segments
```

The cluster that is failing auth shows nothing different:
```
core 2021-07-05T20:19:07Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="4a41369c-21e6-4779-927a-45025a9d90c8"]: failed to verify secret, username: robot$team-otomi+kubernetes, error: failed to get oidc user info, error: <QuerySeter> no row found                                                                                                  │
core 2021-07-05T20:19:07Z [ERROR] [/pkg/token/token.go:66]: parse token error, token contains an invalid number of segments
```
The client that fails:
```
Error response from daemon: login attempt to https://harbor.xxxxxxx/v2/ failed with status: 401 Unauthorized
```

**Comment by sharkymcdongles** [comment]
Ya this is what is most frustrating. It works most of the time but then
randomly fails with the header issue.

**Comment by Morriz** [comment]
Let me also diff the logs in the registry for completeness:

The failing registry:
```
time="2021-07-05T20:19:08.125619711Z" level=error msg="error authenticating user "admin": authentication failure" go.version=go1.15.12 http.request.host=harbor.xxxxx http.request.id=d02c594b-1180-41fb-8dd1-ee633a1fd7a3 http.request.method=GET http.request.remoteaddr=127.0.0.1 http.request.uri="/v2/" http.request.useragent="docker/20.10.6 go/go1 │
time="2021-07-05T20:19:08.125697914Z" level=warning msg="error authorizing context: basic authentication challenge for realm "harbor-registry-basic-realm": authentication failure" go.version=go1.15.12 http.request.host=harbor.xxxxx http.request.id=d02c594b-1180-41fb-8dd1-ee633a1fd7a3 http.request.method=GET http.request.remoteaddr=127.0.0.1 htt │
```

The working registry:
```
127.0.0.6 - - [05/Jul/2021:20:24:36 +0000] "GET /v2/ HTTP/1.1" 200 2 "" "docker/20.10.6 go/go1.13.15 git-commit/8728dd2 kernel/5.10.25-linuxkit os/linux arch/amd64 UpstreamClient(Docker-Client/20.10.6 \\(darwin\\))"
```

**Comment by Morriz** [comment]
Another observation:

On the cluster where I am able to successfully `docker login` with the token, a pullSecret created with the same information is NOT working.

**Comment by sharkymcdongles** [comment]
So I did some more testing and investigation. It seems the auth header issues and errors were coming from the ingress nginx. Ingress nginx doesn't allow _ in a header by default. After enabling:

`enable-underscores-in-headers: 'true'`

I was able to run a while loop pulling and removing an image with no failures. That being said I still see NOT_FOUND errors in the logs constantly despite the image existing. I will dig deeper on this issue as it may be unrelated.

examples:

```
2021-07-06T10:35:03Z [DEBUG] [/lib/http/error.go:59]: {"errors":[{"code":"NOT_FOUND","message":"blob sha256:4e006334a6fdea37622f72b21eb75fe1484fc4f20ce8b8526187d6f7bd90a6fe not found"}]}
2021-07-06T10:35:03Z [DEBUG] [/lib/http/error.go:59]: {"errors":[{"code":"NOT_FOUND","message":"blob sha256:cdc9dae211b46c95cfcf70f8c98aa6ad0e114439cb5ce2f229b078338f0a4f64 not found"}]}
2021-07-06T10:35:03Z [DEBUG] [/lib/http/error.go:59]: {"errors":[{"code":"NOT_FOUND","message":"blob sha256:fe6a4fdbedc0cbc560437fa700b3b034114e31a264f0818d0d32ed2ee6cbe7a3 not found"}]}
2021-07-06T10:35:03Z [DEBUG] [/lib/http/error.go:59]: {"errors":[{"code":"NOT_FOUND","message":"blob sha256:e4d0e810d54a9c47e3dface412b0b88045d156b18809a2e0bbfb0fc8a45d8127 not found"}]}
2021-07-06T10:35:03Z [DEBUG] [/lib/http/error.go:59]: {"errors":[{"code":"NOT_FOUND","message":"blob sha256:4b4c002ee6ca02473de9ed967cc372717121a986f21ea876f9d854284481a6d0 not found"}]}
2021-07-06T10:35:03Z [DEBUG] [/lib/http/error.go:59]: {"errors":[{"code":"NOT_FOUND","message":"blob sha256:7095af798ace32173839a61cbf101048434e1065185c0f29cc888e67158d990b not found"}]}
```

**Comment by sharkymcdongles** [comment]
Seems I still see these after the adjustment made above:

`021-07-06T12:22:07Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="9d5bf08d0598b184a4a9f7ce39f65f0c"]: failed to verify secret, username: robot$gitlab, error: failed to get oidc user info, error: <QuerySeter> no row found`

also still seeing these:

`2021-07-06T12:22:07Z [DEBUG] [/lib/http/error.go:59]: {"errors":[{"code":"UNAUTHORIZED","message":"authorize header needed to send HEAD to repository: authorize header needed to send HEAD to repository"}]}`

**Comment by Morriz** [comment]
We have been spending quite some time on figuring this out. The last two lines you mentioned seem to just be noise because of the software design error ("trying" a resolution path, expecting that to fail, then going to the next strategy). I wish software makers would stray from creating noise when they could easily avoid it.

**Comment by bsinggh** [comment]
Hello, did anyone find any solution for this issue?

I just upgraded from 2.2.1 to 2.3.1 and see same error. I am not able to login to docker using robot account. 

Jul 26 16:22:22  core[2519]: 2021-07-26T16:22:22Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="092sfa74-c11b-4966-92a9-e6alleda2eae"]: failed to verify secret, username: robot$testupgrade, error: failed to get oidc user info, error: <QuerySeter> no row found
Jul 26 16:22:22 core[2519]: 2021-07-26T16:22:22Z [ERROR] [/pkg/token/token.go:66]: parse token error, token contains an invalid number of segments
Jul 26 16:22:22 core[2519]: 2021-07-26T16:22:22Z [ERROR] [/server/middleware/security/robot2.go:47][requestID="092sfa74-c11b-4966-92a9-e6alleda2eae"]: failed to authenticate robot account: robot$testupgrade
Jul 26 16:22:22 core[2519]: 2021-07-26T16:22:22Z [ERROR] [/core/auth/authenticator.go:264]: Failed to get user from DB, username: robot$testupgrade, error: user robot$testupgrade not found

**Comment by whereismyjetpack** [comment]
For me, the upgrade from 2.2 to 2.3 changed the way in which the registry gets it's htpasswd. i was able to remedy this by copying the htpasswd field out of the `-harbor-registry` secret and placing the same value into the `-harbor-registry-htpasswd` secret.

**Comment by sharkymcdongles** [comment]
> For me, the upgrade from 2.2 to 2.3 changed the way in which the registry gets it's htpasswd. i was able to remedy this by copying the htpasswd field out of the `-harbor-registry` secret and placing the same value into the `-harbor-registry-htpasswd` secret.

For my side it seems this is set correctly, but I suspect perhaps the htpasswd implementation doesn't work with OIDC properly, and maybe switching to token based might work. Will experiment and get back here.

**Comment by sharkymcdongles** [comment]
So more digging this error could be solved by adding a check to this method to return nil if the username prefix is the same as the robot prefix: 

`2021-08-24T11:15:42Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="f7b12e7636f6e9fd320dd6e4ac518d77"]: failed to verify secret, username: ROBOTACCOUNT, error: failed to get oidc user info, error: <QuerySeter> no row found`

This would eliminate one of the big error logs we see.

**Comment by DennisGlindhart** [comment]
This issue seems gone for me with 2.3.2 - Maybe issue https://github.com/goharbor/harbor/issues/15290 had something to do with it?

Anyone else can confirm?

**Comment by acobaugh** [comment]
Still seeing this with 2.3.3, at least when doing a `helm chart push` using a robot account.

```
harbor-test-core-fd897d4f8-z9p8s core 2021-09-28T21:36:49Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="eb2db930c17222d5b43cb97637254b1d"]: failed to verify secret, username: robot+helm+atc135, error: failed to get oidc user info, error: <QuerySeter> no row found
harbor-test-core-fd897d4f8-z9p8s core 2021-09-28T21:36:49Z [INFO] [/server/middleware/security/robot2.go:74][requestID="eb2db930c17222d5b43cb97637254b1d"]: a robot2 security context generated for request GET /service/token
```

The client then shows:

```
Error: unexpected response: 401 Unauthorized
```

**Comment by bsinggh** [comment]
Did anyone resolve this? I am seeing same issue in 2.3.4 version. It still thinks robot user as oidc user and tries to get info from DB.

Nov 12 18:16:19  core[2641]: 2021-11-12T18:16:19Z [ERROR] [/server/middleware/security/robot2.go:47][requestID="9ded48d7-5d7b-44350-aeb9-4b074796de2"]: failed to authenticate robot account: robot$testupgrade

Nov 12 18:16:19  core[2641]: 2021-11-12T18:16:19Z [ERROR] [/core/auth/authenticator.go:262]: Failed to get user from DB, username: robot$testupgrade, error: user robot$testupgrade not found

Nov 12 18:16:19 [2641]: 2021-11-12T18:16:19Z [ERROR] [/server/middleware/security/basic_auth.go:40][requestID="9ded48d7-5d7b-44350-aeb9-4b074796de2"]: failed to authenticate robot$testupgrade: not supported

**Comment by bsinggh** [comment]
Did anyone resolve this? I am seeing below for 2.3.4 harbor version.

Nov 12 18:25:54  core[2641]: 2021-11-12T18:25:54Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="07a74387-4cc5-4d94-bde9-b3dge0509939"]: failed to verify secret, username: robot$testupgrade, error: failed to get oidc user info, error: <QuerySeter> no row found

Nov 12 18:25:54  core[2641]: 2021-11-12T18:25:54Z [ERROR] [/pkg/token/token.go:66]: parse token error, token contains an invalid number of segments

Nov 12 18:25:54  core[2641]: 2021-11-12T18:25:54Z [INFO] [/server/middleware/security/robot2.go:74][requestID="07a74387-4cc5-4d94-bde9-b3dge0509939"]: a robot2 security context generated for request GET /service/token

**Comment by j-zimnowoda** [comment]
:(

**Comment by yogeek** [comment]
Same error with 2.4 harbor version (harbor installed with helm, oidc with dex)

**Comment by adberger** [comment]
Not working for us either with Version v2.4.0-d4affc2e

`2021-12-27T16:07:58Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="db599dc81b53bfb8b80d0e1816bd4757"]: failed to verify secret, username: robot$project+helm-push, error: failed to get oidc user info, error: <QuerySeter> no row found`

**Comment by Quantum-Sicarius** [comment]
Same issue:
Version v2.4.1-c4b06d79

```
Dec 31 07:58:03 172.19.0.1 core[2343]: 2021-12-31T07:58:03Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="0a5dd53c-531c-47e9-a446-ea3af701d903"]: failed to verify secret, username: robot$up-node, error: failed to get oidc user info, error: <QuerySeter> no row found
Dec 31 07:58:03 172.19.0.1 core[2343]: 2021-12-31T07:58:03Z [ERROR] [/server/middleware/security/basic_auth.go:40][requestID="0a5dd53c-531c-47e9-a446-ea3af701d903"]: failed to authenticate robot$up-node: not supported
```

**Comment by Quantum-Sicarius** [comment]
After further investigation, it would appear that Legacy robot accounts appear to be the actual issue as a new robot account works with OIDC.

Version:
 Version v2.4.1-c4b06d79

**Comment by adriannieto-attechnest** [comment]
We upgraded from 1.8 to latest version and we are in the same situation. This happens in automated robot accounts like trivy one too. 

````harbor-core-68f446c9bf-pjm8v core 2022-02-04T06:55:09Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="1a3d6e25-9104-44f7-99a7-1dc56a507c62"]: failed to verify secret, username: robot$xxx+Trivy-cbdab39d-8585-11ec-aabe-765eaaf3b54e, error: failed to get oidc user info, error: <QuerySeter> no row found```

**Comment by j14s** [comment]
My logs are getting absolutely flooded right now with OIDC errors like this.

**Comment by sremp** [comment]
We've been receiving these same error messages for some time now, however our GitHub Actions image publishing was working up until last Thursday when our Harbor Helm release auto-updated. Now, all we are seeing is:
`[ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="xxxxxxxx"]: failed to verify secret, username: xxxx, error: failed to get oidc user info, error: <QuerySeter> no row`

**Comment by warmfusion** [comment]
> After further investigation, it would appear that Legacy robot accounts appear to be the actual issue as a new robot account works with OIDC.
> 
> Version: Version v2.4.1-c4b06d79

We're also using Version v2.4.1-c4b06d79 but a robot account created on the 31st March (6 days ago) shows the same error.

We're using an `@` rather than `$` for the robot separator, have been upgrading harbor since the 1.x era so there may be something with an upgrade process causing issues but its not clear.

**Comment by eg-ops** [comment]
In our case "Garbage Collection" in the settings tab fixed the problem. The main problem seems to be in running out of memory in minio service.

**Comment by zdhamasha** [comment]
any updates regarding to this issue ?

**Comment by vwatinteg** [comment]
Same as zdhamasha, any updates? We're on
CHART NAME: harbor
CHART VERSION: 11.1.0
APP VERSION: 2.4.0

Been getting these logs for months...

**Comment by IDerr** [comment]
I'm getting these logs too :(

**Comment by naitmare01** [comment]
Same for me.

**Comment by nedhanks** [comment]
I'm getting these errors on a fresh install of 2.5.3. OIDC is configured, and seems to be work ok. Created a robot account to do replication from another harbor install and seeing a lot of these errors. 

```
NAME  	NAMESPACE    	REVISION	UPDATED                             	STATUS  	CHART       	APP VERSION
marina	harbor-marina	1       	2022-07-29 19:28:33.161549 -0600 MDT	deployed	harbor-1.9.3	2.5.3
```


Also, I'm unable to login using docker, but I can login to the UI. I copied the token from my profile.
```
docker login marina.[redacted] -u ned_hanks -p v5HIeUuUx
WARNING! Using --password via the CLI is insecure. Use --password-stdin.
Error response from daemon: Get "https://marina.octanner.io/v2/": unauthorized: authentication required
```
```
2022-08-12T23:27:03Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="03bc7f14-0fa2-4eda-9c16-eb1d9877043a"]: failed to verify secret, username: ned_hanks, error: failed to get oidc user info, error: <QuerySeter> no row found
2022-08-12T23:27:03Z [ERROR] [/server/middleware/security/basic_auth.go:72][client IP="10.2.48.0, 10.2.9.212" requestID="03bc7f14-0fa2-4eda-9c16-eb1d9877043a" user agent="docker/20.10.12 go/go1.16.12 git-commit/459d0df kernel/5.10.76-linuxkit os/linux arch/amd64 UpstreamClient(Docker-Client/20.10.12 \(darwin\))"]: failed to authenticate user:ned_hanks, error:not supported
```

**Comment by adriannieto-attechnest** [comment]
@wy65701436 could you delete Ned comment? He is leaking credentials by mistake

**Comment by danielfm** [comment]
I'm experiencing this issue too.

Using a robot account in `helm repo add` command fails, but the same account works when using `helm registry login`.

```
$ helm registry login <url> --username 'robot$<user>' --password '<token>'
WARNING: Using --password via the CLI is insecure. Use --password-stdin.
Login Succeeded

$ helm repo add test <url>/chartrepo --username 'robot$<user>' --password '<token>'
Error: looks like "<url>" is not a valid chart repository or cannot be reached: failed to fetch <url>/chartrepo/index.yaml : 403 Forbidden
```

**Comment by jimsnab** [comment]
I hit this yesterday on a fresh install. I have Jenkins build automation that uploads build results to Harbor. I searched all day yesterday to try and figure out why I got this `no row found` error. Then inexplicably it worked, I did nothing but re-run the build in attempts to find more details in logs.

**Comment by cesarb1392** [comment]
I'm still running into this issue. Deploying Harbor using the Helm chart version v1.10.2

```
[ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="552f33a7-4a03-4b04-8782-f5aa6152b578"]: failed to verify secret, username: robot$account, error: failed to get oidc user info, error: <QuerySeter> no row found
[INFO] [/server/middleware/security/robot.go:71][requestID="552f33a7-4a03-4b04-8782-f5aa6152b578"]: a robot security context generated for request GET /service/token
```

**Comment by steadyk** [comment]
I also see this on v1.10.2, but with slightly different errors:

```
2022-11-21T07:56:55Z [ERROR] [/server/middleware/security/basic_auth.go:72][client IP="10.244.0.70:55126" requestID="4251fe16-8e20-4c74-ad08-774836c87951" user agent="curl/7.72.0-DEV"]: failed to authenticate user:admin, error:Failed to authenticate user, due to error 'Invalid credentials'
2022-11-21T07:56:57Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="70a43bf6-3933-4c4a-b79d-9b93ab2a4485"]: failed to verify secret, username: admin, error: failed to get oidc user info, error: <QuerySeter> no row found
2022-11-21T07:56:57Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="3eb4527f-1039-463f-ba57-4d2f10a5122c"]: failed to verify secret, username: admin, error: failed to get oidc user info, error: <QuerySeter> no row found
2022-11-21T07:56:57Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="72f15e3f-8f9d-45c6-8e5e-e257e4e8f216"]: failed to verify secret, username: admin, error: failed to get oidc user info, error: <QuerySeter> no row found
```

Mabye that also happens due to migration from v.1.10.1 -> v1.10.2.

**Comment by rgarcia89** [comment]
Any update here? I am also see the core pod logs getting filled with that

**Comment by devlifealways** [comment]
Same here, waiting for this fix to be out once 
```shell
2023-01-30T13:21:56.445777863Z 2023-01-30T13:21:56Z [INFO] [/jobservice/logger/service.go:63]: Found export data cleanup job with schedule id : 1
2023-01-30T13:22:56.435408139Z 2023-01-30T13:22:56Z [INFO] [/pkg/notifier/notifier.go:205]: Handle notification with Handler 'AuditLog' on topic 'CREATE_PROJECT': ID-2 Name-gitlab Operator-hrouineb OccurAt-2023-01-30 13:22:56
2023-01-30T13:24:57.636334678Z 2023-01-30T13:24:57Z [INFO] [/controller/registry/controller.go:222]: Start regular health check for registries with interval 5m0s
2023-01-30T13:26:03.476178027Z 2023-01-30T13:26:03Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="2115c3f2-a75e-4742-aeed-83d52c917773"]: failed to verify secret, username: robot$gitlab+yinn, error: failed to get oidc user info, error: <QuerySeter> no row found
2023-01-30T13:26:03.483522200Z 2023-01-30T13:26:03Z [INFO] [/server/middleware/security/robot.go:71][requestID="2115c3f2-a75e-4742-aeed-83d52c917773"]: a robot security context generated for request GET /service/token
2023-01-30T13:28:44.937218308Z 2023-01-30T13:28:44Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="15cd5e03-4df9-4d74-83b9-9a01eec5de8d"]: failed to verify secret, username: robot$gitlab+yinn, error: failed to get oidc user info, error: <QuerySeter> no row found
2023-01-30T13:28:44.947952864Z 2023-01-30T13:28:44Z [INFO] [/server/middleware/security/robot.go:71][requestID="15cd5e03-4df9-4d74-83b9-9a01eec5de8d"]: a robot security context generated for request GET /service/token
2023-01-30T13:33:43.333409691Z 2023-01-30T13:33:43Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="206acf81-b59b-4564-b647-91f55b3bba86"]: failed to verify secret, username: robot$gitlab+yinn, error: failed to get oidc user info, error: <QuerySeter> no row found
2023-01-30T13:33:43.342219923Z 2023-01-30T13:33:43Z [INFO] [/server/middleware/security/robot.go:71][requestID="206acf81-b59b-4564-b647-91f55b3bba86"]: a robot security context generated for request GET /service/token
2023-01-30T13:34:02.902550312Z 2023-01-30T13:34:02Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="487cec82-70cf-438f-9638-11b3fe2d5eb4"]: failed to verify secret, username: robot$gitlab+yinn, error: failed to get oidc user info, error: <QuerySeter> no row found
2023-01-30T13:34:02.909669962Z 2023-01-30T13:34:02Z [INFO] [/server/middleware/security/robot.go:71][requestID="487cec82-70cf-438f-9638-11b3fe2d5eb4"]: a robot security context generated for request GET /service/token
2023-01-30T13:38:45.055944272Z 2023-01-30T13:38:45Z [ERROR] [/server/middleware/security/oidc_cli.go:62][requestID="95eb830b-e4df-4929-9700-dd5d9e5ea436"]: failed to verify secret, username: robot$hisqool, error: failed to get oidc user info, error: <QuerySeter> no row found
2023-01-30T13:38:45.063001194Z 2023-01-30T13:38:45Z [INFO] [/server/middleware/security/robot.go:71][requestID="95eb830b-e4df-4929-9700-dd5d9e5ea436"]: a robot security context generated for request GET /service/token

```

**Comment by devlifealways** [comment]
A misleading issue, the problem was not related to OIDC authentication, it was all about registry basic auth wrong credentials I was using.

**Comment by umberto10** [comment]
So is the fix available in version 2.7.1? Because I'm still having this issue in my latest helm install :<

**Comment by rgarcia89** [comment]
> So is the fix available in version 2.7.1? Because I'm still having this issue in my latest helm install :<

This fix will be included in v2.8.0

**Comment by ryadama9** [comment]
Is there any work around without moving to 2.8.0? we are in 2.6.1

**Comment by nerzhul** [comment]
I have same issue on harbor 2.9 with admin local user
```
username: admin, error: failed to get oidc user info, error: <QuerySeter> no row found
```

**Comment by EsDmitrii** [comment]
the same

**Comment by yelassad** [comment]
the same on 2.7.3

**Comment by steadyk** [comment]
Harbor 2.9.1
We also see errors for admin user:

```
2023-11-15T09:38:07Z [ERROR] [/server/middleware/security/oidc_cli.go:68][requestID="b260d299-612e-4bae-a771-bed7155e1767"]: failed to verify secret, username: admin, error: failed to get oidc user info, error: <QuerySeter> no row found
```

**Comment by peresureda** [comment]
Harbor 2.9.1 Upgrade:

`2023-11-16T17:08:36Z [ERROR] [/server/middleware/security/oidc_cli.go:68][requestID="xxxxx"]: failed to verify secret, username: admin, error: failed to get oidc user info, error: <QuerySeter> no row found`

**Comment by pwurbs** [comment]
Maybe it helps someone:
- We face the "<QuerySeter> no row found" error message all the time. It doesn't mean anything.
- We faced "docker login" 401 errors too. In our case the root cause was a mismatch of REGISTRY_HTPASSWD and REGISTRY_PASSWD in our K8S secret

**Comment by KevinGimbel** [comment]
We face the same issue with v2.10.1

```
2024-04-06T05:01:16Z [ERROR] [/server/middleware/security/oidc_cli.go:68][requestID=xxx]: failed to verify secret, username: admin, error: failed to get oidc user info, error: <QuerySeter> no row found
```

**Comment by sicko583** [comment]
I had the same issue with v.2.10.2

```
2024-06-20T03:07:55Z [ERROR] [/server/middleware/security/oidc_cli.go:68][requestID="334fcd11abfd96644d485b9ab971f268"]: failed to verify secret, username: robot, error: failed to get oidc user info, error: <QuerySeter> no row found
2024-06-20T03:07:55Z [ERROR] [/server/middleware/security/basic_auth.go:72][client IP="10.27.63.71" requestID="334fcd11abfd96644d485b9ab971f268" user agent="docker/1.13.1 go/go1.10.3 kernel/5.10.16.3-microsoft-standard-WSL2 os/linux arch/amd64 UpstreamClient(Docker-Client/1.13.1 \(linux\))"]: failed to authenticate user:robot, error:not supported
```

Refer to https://github.com/goharbor/harbor/issues/20629

**Comment by peters-david** [comment]
We are facing this issue also on v2.10.2.

**Comment by rajatrj16** [comment]
We are facing this issue on 2.11.0
```
failed to verify secret, username: admin, error: failed to get oidc user info, error: <QuerySeter> no row found
```
However, we have local system admin user, OIDC users for real users and robot accounts for pipeline tasks.

**Comment by KevinGimbel** [comment]
Still happening with `Version v2.11.1-6b7ecba1`. Our alerting just fired due to harbor logging a lot of these lines:

```
2024-12-03T19:10:06Z [ERROR] [/server/middleware/security/oidc_cli.go:68][requestID="x-x-x-x-x"]: failed to verify secret, username: admin, error: failed to get oidc user info, error: <QuerySeter> no row found
```

**Comment by peters-david** [comment]
This issue still persists. Is it possible to reopen? @stonezdj

**Comment by MalteMagnussen** [comment]
Still happening @stonezdj

**Comment by rajatrj16** [comment]
We are still facing this issue on 2.12.4

`failed to verify secret, username: admin, error: failed to get oidc user info, error: <QuerySeter> no row found`

However, we have local system admin user, OIDC users for real users and robot accounts for pipeline tasks.


**While I looked at issues, I see above is fixed and available in 2.13.0** 
https://github.com/goharbor/harbor/pull/21061
